File: src/main/java/by/masarnovsky/releasedemon/service/UserService.java
package by.masarnovsky.releasedemon.service;

import by.masarnovsky.releasedemon.entity.User;

import java.util.List;

public interface UserService {
    User findByLogin(String login);

    User findById(Integer id);

    List<User> findAllTelegramUsers();
}

File: src/main/kotlin/by/masarnovsky/releasedemon/service/UserService.kt
package by.masarnovsky.releasedemon.service

import by.masarnovsky.releasedemon.entity.User
import by.masarnovsky.releasedemon.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(
    val repository: UserRepository,
) {
    fun findByLogin(login: String): User? {
        return repository.findByLogin(login)
    }

    fun findAllTelegramUsers(): List<User> {
        return repository.findAllByTelegramIdNotNull()
    }
}